src/link/Elf.zig

```diff
@@ -1505,13 +1505,10 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
     }
 
     if (is_dyn_lib) {
-        const soname = self.base.options.override_soname orelse if (self.base.options.version) |ver|
-            try std.fmt.allocPrint(arena, "lib{}.so.{}", .{ self.base.options.root_name, ver.major })
-        else
-            try std.fmt.allocPrint(arena, "lib{}.so", .{self.base.options.root_name});
-        try argv.append("-soname");
-        try argv.append(soname);
-
+        if (self.base.options.override_soname) |soname| {
```

Comment:
Should `override_soname` get renamed to just `soname` then?

Reply:
You're probably right, and that's exactly what I started doing but it's littered around the place (link.zig, main.zig, Compilation.zig, Elf.zig, glibc.zig, etc.) so I held it off. But since you believe it should be done just like me, I'm eager to refactor the necessary bits :-)

Reply:
(or `explicit_soname` ?)

Reply:
I think if we're going to change it, `soname` should be fine.